src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -593,6 +593,49 @@ public void FunctionPrecedenceTest(string script, double expected)
             Assert.Equal(expected, result.AsDouble());
         }
 
+        [Theory]
+
+        // Return value with side effectful UDF
+        [InlineData(
+            "F1(x:Number) : Number = { Set(a, x); a+1; };",
+            "F1(123)",
+            false,
+            124,
+            true)]
+
+        // Mismatch return value with side effectful UDF
+        [InlineData(
+            "F1(x:Number) : Boolean = { Set(a, x); Today(); };",

```

Comment:
>"F1(x:Number) : Boolean = { Set(a, x); Today(); }; [](http://example.com/codeflow?start=12&length=50)

Where do you expect this to throw? During AddUserDefinedFunctio, or during Eval? It would be good if the test could tell us that.

Reply:
It throws during AddUserDefinedFunction, `CheckTypesOnDeclaration`